tools/clustinfo.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+"""
+Module to contain information about running clusters to use for testing.
+Specifically:
+ Access URL
+ auth token
+ node count
+ whether they were auto-created during test invocation
+"""
+
+import logging
+import os
+import time
+
+import launch_ccm_cluster
+
+logger = logging.getLogger(__name__)
+
+# holds info objects
+_clusters = []
+
+def start_cluster(launch_config=None, reporting_name=None):
+    if not reporting_name:
+        reporting_name="only_cluster"
+
+    _launch_recorder.start(reporting_name)
+    try:
+        cluster = _launch_cluster(launch_config)
+        _launch_recorder.finish_ok(reporting_name, cluster)
+    except:
+        _launch_recorder.finish_fail(reporting_name)
+        raise
+    _clusters.append(cluster)
+    logger.info("Started cluster: %s", cluster.cluster_id)
+    return cluster
+
+def get_launch_attempts():
+    return _launch_recorder.get_list()
+
+def add_running_cluster(url, auth_token):
+    cluster = ClusterInfo(url, auth_token, external=True)
+    _clusters.append(cluster)
+    return cluster
+
+def get_cluster_by_url(cluster_url):
+    matches = [cluster for cluster in _clusters if cluster.url == cluster_url]
+    assert len(matches) in (0, 1)
```

Comment:
I chose this to explicitly emphasize that 0 was valid, partly because I felt the list comprehension was misleading.  It might be better off just returning the first match or None.